lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java

```diff
@@ -209,19 +212,39 @@ public void clear() {
 	}
 
 	public List<SimplePageItem> findItemsInSite(String siteId) {
-	    Object [] fields = new Object[1];
-	    fields[0] = siteId;
-	    List<String> ids = sqlService.dbRead("select b.id from lesson_builder_pages a,lesson_builder_items b,SAKAI_SITE_PAGE c where a.siteId = ? and a.parent is null and a.pageId = b.sakaiId and b.type = 2 and b.pageId = 0 and a.toolId = c.PAGE_ID order by c.SITE_ORDER", fields, null);
+		// This is a three-step process.
+		// 1) Find the pages in the site
+		// 2) Find the Lessons pages 
+		// 3) Find the Lessons items
+		try {
+			Site site = siteService.getSite(siteId);
+			List<SitePage> sitePages = site.getOrderedPages();
 
-	    List<SimplePageItem> result = new ArrayList<SimplePageItem>();
-	    
-	    if (result != null) {
-		for (String id: ids) {
-		    SimplePageItem i = findItem(new Long(id));
-		    result.add(i);
+			List<String> sitePageIds = new ArrayList<>();
+			for (SitePage page : sitePages) {
+				sitePageIds.add(page.getId());
+			}
+
+			DetachedCriteria d = DetachedCriteria.forClass(SimplePage.class);
+			d.add(Restrictions.in("toolId", sitePageIds));
+			d.add(Restrictions.isNull("parent"));
+
+			List<SimplePage> lessonsPages = (List<SimplePage>) getHibernateTemplate().findByCriteria(d);
+			List<String> lessonsPageIds = new ArrayList<>();
+			if (lessonsPages != null && lessonsPages.size() > 0) {
```

Comment:
You could use CollectionUtils.isEmpty(list) here

Reply:
Or just `!lessonsPages.isEmpty()` as I don't think `findByCriteria`should return `null`